src/commands/account/delete_key/mod.rs

```diff
@@ -26,23 +28,36 @@ impl DeleteKeyCommandContext {
         Ok(Self {
             config: previous_context.0,
             owner_account_id: scope.owner_account_id.clone().into(),
-            public_key: scope.public_key.clone().into(),
+            public_keys: scope
+                .public_keys
+                .clone()
+                .0
+                .into_iter()
+                .map(Into::into)
+                .collect(),
         })
     }
 }
 
 impl From<DeleteKeyCommandContext> for crate::commands::ActionContext {
     fn from(item: DeleteKeyCommandContext) -> Self {
+        let public_keys_clone = std::sync::Arc::new(item.public_keys.clone());
```

Comment:
Drop the Arc, cloning is enough.

```suggestion
        let public_keys = item.public_keys.clone();
```